python/test/unit/language/test_core.py

```diff
@@ -1329,8 +1329,11 @@ def test_cast(dtype_x, dtype_z, bitcast, size, num_ctas, device):
         if dtype_z in uint_dtypes:
             x = np.absolute(x)
         x_tri = to_triton(x, device=device)
-
+    if 'float' in dtype_z and 'float' in dtype_x:
+        # make sure we use values that can be represented in both types
+        x_tri = x_tri.to(getattr(torch, dtype_z)).to(getattr(torch, dtype_x))
```

Comment:
@ThomasRaoux Does this look fine to you ? It's just a way to avoid mismatches due to precision loss when doing high to low precision casting, e.g. f32->f8, I want to start with an `x` that can be represented in f8, so I do these 2 conversions.